File: mobile/library/swift/src/StatsClient.swift
import Foundation

/// Client used to record time series metrics.
///
/// Note: this an experimental interface and is subject to change. The implementation has not been
/// optimized, and there may be performance implications in production usage.
@objc
public protocol StatsClient: AnyObject {
  /// - parameter elements: Elements to identify a counter
  ///
  /// - returns: A Counter based on the joined elements.
  func counter(elements: [Element]) -> Counter

  /// - parameter elements: Elements to identify a gauge
  ///
  /// - returns: A Gauge based on the joined elements.
  func gauge(elements: [Element]) -> Gauge
}

File: mobile/library/kotlin/src/io/envoyproxy/envoymobile/StatsClient.kt
package io.envoyproxy.envoymobile

/**
 * Client used to record time series metrics.
 *
 * Note: this is an experimental interface and is subject to change The implementation has not been
 * optimized, and there may be performance implications in production usage.
 */
interface StatsClient {

  /**
   * @return A counter based on the joined elements.
   */
  fun counter(vararg elements: Element): Counter

  /** @return A gauge based on the joined elements. */
  fun gauge(vararg elements: Element): Gauge
}
